src/kingdom/AbstractAnimal.java

```diff
@@ -0,0 +1,23 @@
+package kingdom;
+
+public abstract class AbstractAnimal implements Animal {
+
+    public static int id = 0;
+    protected int energy;
+
```

Comment:
If you wanted to implement an incrementing ID here, then you're going to need another variable (perhaps name it something like `nextID`). Make `nextID` static and *don't make `id` static* (so that it won't be the same ID for all instances).

Once this is done, change your default constructor (the one with no parameters) to use `nextID`'s value instead when assigning the value for `this.id`.